tests/statistics_test.py

```diff
@@ -0,0 +1,239 @@
+from hazelcast.config import ClientConfig, ClientProperties, NearCacheConfig
+from hazelcast.core import CLIENT_VERSION, CLIENT_TYPE
+from tests.hzrc.ttypes import Lang
+from tests.util import random_string
+
+
+class StatisticsTest(HazelcastTestCase):
+
+    DEFAULT_STATS_PERIOD = 3
+    STATS_PERIOD = 1
+
+    @classmethod
+    def setUpClass(cls):
+        cls.rc = cls.create_rc()
+        cls.cluster = cls.create_cluster(cls.rc)
+        cls.member = cls.cluster.start_member()
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.rc.exit()
+
+    def test_statistics_disabled_by_default(self):
+        client = HazelcastClient()
+        time.sleep(2 * self.DEFAULT_STATS_PERIOD)
+        response = self._get_client_stats_from_server()
+
+        self.assertTrue(response.success)
+        self.assertIsNone(response.result)
+        client.shutdown()
+
+    def test_statistics_disabled_with_wrong_value(self):
+        config = ClientConfig()
+        config.set_property(ClientProperties.STATISTICS_ENABLED.name, "truee")
+        config.set_property(ClientProperties.STATISTICS_PERIOD_SECONDS.name, self.STATS_PERIOD)
+        client = HazelcastClient(config)
+        time.sleep(2 * self.STATS_PERIOD)
+        response = self._get_client_stats_from_server()
+
+        self.assertTrue(response.success)
+        self.assertIsNone(response.result)
+        client.shutdown()
+
+    def test_statistics_enabled(self):
+        config = ClientConfig()
+        config.set_property(ClientProperties.STATISTICS_ENABLED.name, True)
+        client = HazelcastClient(config)
+        time.sleep(2 * self.DEFAULT_STATS_PERIOD)
+        response = self._get_client_stats_from_server()
+
+        self.assertTrue(response.success)
+        self.assertIsNotNone(response.result)
+        client.shutdown()
+
+    def test_statistics_enabled_with_environment_variable(self):
+        environ = os.environ
+        environ[ClientProperties.STATISTICS_ENABLED.name] = "true"
+        environ[ClientProperties.STATISTICS_PERIOD_SECONDS.name] = str(self.STATS_PERIOD)
+
+        client = HazelcastClient()
+        time.sleep(2 * self.STATS_PERIOD)
+        response = self._get_client_stats_from_server()
+
+        self.assertTrue(response.success)
+        self.assertIsNotNone(response.result)
+        os.unsetenv(ClientProperties.STATISTICS_ENABLED.name)
+        os.unsetenv(ClientProperties.STATISTICS_PERIOD_SECONDS.name)
+        client.shutdown()
+
+    def test_statistics_period(self):
+        config = ClientConfig()
+        config.set_property(ClientProperties.STATISTICS_ENABLED.name, True)
+        config.set_property(ClientProperties.STATISTICS_PERIOD_SECONDS.name, self.STATS_PERIOD)
+        client = HazelcastClient(config)
+
+        time.sleep(2 * self.STATS_PERIOD)
+        response1 = self._get_client_stats_from_server()
+        self.assertTrue(response1.success)
+        self.assertIsNotNone(response1.result)
+
+        time.sleep(2 * self.STATS_PERIOD)
```

Comment:
time dependent test may fail, better to use `eventually` logic (e.g. `assertTrueEventually`).

Reply:
Okay, I will change it